hera_cal/abscal.py

```diff
@@ -0,0 +1,484 @@
+        self.Nants = len(self.ants)
+
+        # setup baselines and antenna positions
+        self.antpos = antpos
+        self.bls = None
+        if self.antpos is not None:
+            self.bls = odict([(x, self.antpos[x[1]] - self.antpos[x[0]]) for x in self.keys])
+            self.antpos = np.array(map(lambda x: self.antpos[x], self.ants))
+            self.antpos -= np.median(self.antpos, axis=0)
+
+    def amp_logcal(self, separate_pol=False, verbose=True):
+        """
+        call abscal_funcs.amp_logcal() method. see its docstring for more details.
+
+        Parameters:
+        -----------
+        separate_pol : bool, separate polarization to have independent solutions
+            if False, form a joint solution across both pols
+
+        Result:
+        -------
+        per-antenna amplitude and per-antenna amp gains
+        can be accessed via the get functions
+            self.ant_eta
+            self.ant_eta_gain
+        """
+        # set data quantities
+        model = self.model
+        data = self.data
+        wgts = self.wgts
+
+        if separate_pol:
+            model, pols = data_key_to_array_axis(model, 2)
+            data, pols = data_key_to_array_axis(data, 2)
+            wgts, pols = data_key_to_array_axis(wgts, 2)
+
+        # run linsolve
+        fit = amp_logcal(model, data, wgts=wgts, verbose=verbose)
+
+        # form result array
+        self._ant_eta = np.array(map(lambda a: fit['eta{}'.format(a)], self.ants))
+
+    def phs_logcal(self, separate_pol=False, verbose=True):
+        """
+        call abscal_funcs.phs_logcal() method. see its docstring for more details.
+
+        Parameters:
+        -----------
+        separate_pol : bool, separate polarization to have independent solutions
+            if False, form a joint solution across both pols
+
+        Result:
+        -------
+        per-antenna phase and per-antenna phase gains
+        can be accessed via the methods
+            self.ant_phi
+            self.ant_phi_gain
+        """
+        # assign data
+        model = self.model
+        data = self.data
+        wgts = self.wgts
+
+        # separate pol
+        if separate_pol:
+            model, pols = data_key_to_array_axis(model, 2)
+            data, pols = data_key_to_array_axis(data, 2)
+            wgts, pols = data_key_to_array_axis(wgts, 2)
+
+        # run linsolve
+        fit = phs_logcal(model, data, wgts=wgts, verbose=verbose)
+
+        # form result array
+        self._ant_phi = np.array(map(lambda a: fit['phi{}'.format(a)], self.ants))
+
+    def delay_lincal(self, medfilt=True, kernel=(1, 11), time_ax=0, freq_ax=1, verbose=True, time_avg=False):
+        """
+        Solve for per-antenna delay according to the equation
+        by calling abscal_funcs.delay_lincal method.
+        See abscal_funcs.delay_lincal for details.
+
+        Currently only supports joint polarization solutions.
```

Comment:
We definitely don't want join delays. We only want x delays and y delays, but they are not in general the same.

It's OK to support just 1pol and 2pol and not 4pol here. The cross-polarized visibilities are not doing any work here. If the input data is 4pol, just use it in 2pol mode.